extensions/emmet/src/matchTag.ts

```diff
@@ -37,6 +40,19 @@ function getUpdatedSelections(editor: vscode.TextEditor, position: vscode.Positi
 		return;
 	}
 
+	// check if this node is a script representing an HTML template. If so, parse the inside as HTML and select appropriate node
+	if (currentNode.name === 'script' &&
+		(position.isAfter(currentNode.open.end) && position.isBefore(currentNode.close.start)) &&
```

Comment:
Also this whole logic of determining whether current node is a script tag with type as text and one of the "allowedMimeTypesInScriptTag" as value should be moved to the `util.ts` file because it is being used more than once.